packages/paste-core/components/badge/src/utils.ts

```diff
@@ -0,0 +1,66 @@
+import type {TextColorOptions, BackgroundColorOptions} from '@twilio-paste/style-props';
+
+import type {BadgeVariants, BadgeProps} from './types';
+
+export const hasValidButtonVariantProps = ({as, onClick}: Pick<BadgeProps, 'as' | 'onClick'>): boolean => {
+  const clickHandlerIsValid = typeof onClick == 'function';
+  const asPropIsValid = as === 'button';
+  return clickHandlerIsValid && asPropIsValid;
+};
+export const hasValidAnchorVariantProps = ({as, href}: Pick<BadgeProps, 'as' | 'href'>): boolean => {
+  const asPropIsValid = as === 'a';
+  const hrefIsValid = typeof href === 'string';
+  return asPropIsValid && hrefIsValid;
+};
+
+type Props = Record<string, any>;
+export const safelySpreadProps = (props: Props, denylist: string[]): Partial<Props> =>
+  Object.keys(props).reduce<Props>((resultProps, key: string) => {
+    if (!denylist.includes(key)) {
+      // eslint-disable-next-line no-param-reassign
+      resultProps[key] = props[key];
+    }
+    return resultProps;
+  }, {});
+
+export const getVariantStyles = (
```

Comment:
ultra nit: for style maps, I prefer object maps rather than switches, but this is totally fine.  In the future, you may consider something like:
```
const variantStyles = {
  'success': {
    backgroundColor: 'colorBackgroundSuccessWeakest',
    color: 'colorTextSuccess',
  },
  ..
}

const styles = variantStyles[variant] || variantStyles.default;
```

It's a little more readable and less code to write.